browser/search_engines/search_engine_tracker.cc

```diff
@@ -89,7 +122,9 @@ SearchEngineTracker::SearchEngineTracker(
     const GURL url = template_url->GenerateSearchURL(search_terms);
     if (!url.is_empty()) {
       default_search_url_ = url;
+      previous_search_url_ = url;
       RecordSearchEngineP3A(url, template_url->GetEngineType(search_terms));
+      RecordSwitchP3A(url);
```

Comment:
Thanks for the review! I agree this isn't ideal. I think we need to record some kind of signal when there hasn't been a switch in the P3A window, so there's a baseline for how often switches happen. I also expect they're rare, but it would be nice to have a denominator to tell us how rare.

Maybe we could store the previous default engine in a pref with a timestamp, or extend `WeeklyStorage` and revert to `kNoSwitch` a week after the last switch event?

Reply:
> Maybe we could store the previous default engine in a pref with a timestamp, or extend WeeklyStorage and revert to kNoSwitch a week after the last switch event?

That may be a good idea to use `WeeklyStorage`.

Reply:
would be great to check the exact metric meaning with the metric owner. Perhaps it should be worded like "did you switch ... this week?" or something like that.

Reply:
> would be great to check the exact metric meaning with the metric owner. Perhaps it should be worded like "did you switch ... this week?" or something like that.

"did you switch **this week**" would provide much more useful data.

Reply:
This should work now.